package com.mega.superx.filter.camera.ui.camera

import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.animation.core.tween
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.scale
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun CountdownOverlay(
    countdownValue: Int,
    modifier: Modifier = Modifier
) {
    if (countdownValue > 0) {
        
        val scale by animateFloatAsState(
            targetValue = 1f,
            animationSpec = tween(durationMillis = 300),
            label = "countdownScale"
        )
        
        Box(
            modifier = modifier.fillMaxSize(),
            contentAlignment = Alignment.Center
        ) {
            Text(
                text = countdownValue.toString(),
                fontSize = 120.sp,
                fontWeight = FontWeight.Bold,
                color = Color.White,
                style = TextStyle(shadow = ViewfinderTextShadow),
                modifier = Modifier.scale(scale)
            )
        }
    }
}

@Composable
fun BurstCaptureOverlay(
    count: Int,
    modifier: Modifier = Modifier
) {
    if (count > 0) {
        
        val scale by animateFloatAsState(
            targetValue = 1f,
            animationSpec = tween(durationMillis = 300),
            label = "countdownScale"
        )

        Box(
            modifier = modifier.fillMaxSize(),
            contentAlignment = Alignment.TopCenter
        ) {
            Text(
                text = count.toString(),
                fontSize = 36.sp,
                fontWeight = FontWeight.Bold,
                color = Color.White,
                style = TextStyle(shadow = ViewfinderTextShadow),
                modifier = Modifier.padding(top = 16.dp).scale(scale)
            )
        }
    }
}
